bundle/Core/Pagination/Pagerfanta/RelatedContentAdapter.php

```diff
@@ -54,9 +64,30 @@ public function setTag(Tag $tag)
         $this->tag = $tag;
     }
 
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\SortClause[] $sortClauses
+     */
+    public function setSortClauses(array $sortClauses)
+    {
+        $this->sortClauses = $sortClauses;
+    }
+
+    /**
+     * Sets additional criteria to be used in search.
+     *
+     * @param array $additionalCriteria
+     */
+    public function setAdditionalCriteria(array $additionalCriteria = [])
+    {
+        $this->additionalCriteria = $additionalCriteria;
+    }
+
     /**
      * Returns the number of results.
      *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
```

Comment:
We don't need these here, they're documented on tags service. Besides, this specific method does not throw exceptions by itself.

Same goes for `getSlice` method below.

Reply:
It was due to PHP Storm complaining because the exception is being thrown from TagsService but not being caught in this method so, in theory, `getNbResults` and `getSlice` could throw it as well? But if you think that it's better with it, I've fixed it in https://github.com/netgen/TagsBundle/pull/98/commits/05f7dad27c86fbfdc247c7f08f2966bc63b3a183.

Reply:
Yeah, PHPstorm needs to be configured correctly. Not every thrown exception in the service needs to be caught somewhere.